In [60]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')
pd.set_option('display.max_rows',None)
pd.set_option('display.max_columns',None)

In [61]:
iqibd_df=pd.read_csv('raw/iqibd_properties.csv')
iqibd_df.head()

air_conditioning       area balcony building_type cctv central_heating  \
0              NaN  1380 sqft     NaN     Apartment    Y             NaN   
1              NaN  1470 sqft     NaN     Apartment  NaN             NaN   
2              NaN  1500 sqft     NaN     Apartment    Y             NaN   
3                Y  5100 sqft       Y     Apartment    Y               Y   
4              NaN   710 sqft     NaN     Apartment  NaN             NaN   

  elevator emergency_stairs fitness_center floor  garage garden generator  \
0        Y                Y            NaN   5th     1.0    NaN         Y   
1        Y                Y            NaN   9th     1.0    NaN         Y   
2        Y                Y            NaN   4th     1.0    NaN         Y   
3        Y                Y              Y   6th     4.0      Y         Y   
4      NaN                Y            NaN   5th     NaN    NaN         Y   

  government_gas      interior   land_size      location  num_balcony  \
0            NaN  Un-Furnished   3.5 Katha   Aftabnagar,          4.0   
1            NaN  Un-Furnished     5 Katha       Mirpur,          3.0   
2              Y  Un-Furnished     3 Katha  Mohammadpur,          4.0   
3              Y  Un-Furnished  11.5 Katha    Baridhara,          4.0   
4              Y  Un-Furnished    12 Katha  Mohammadpur,          2.0   

   num_bath_rooms  num_bed_rooms parking pets_allow           price  \
0             3.0            3.0       Y        NaN     ৳9,800,000    
1             3.0            3.0       Y        NaN     ৳9,500,000    
2             4.0            3.0       Y        NaN    ৳10,500,000    
3             6.0            5.0       Y          Y   ৳139,150,000    
4             2.0            2.0     NaN        NaN     ৳4,500,000    

                      property_link reception security servant_room    status  \
0  https://iqibd.com/properties/306         Y        Y          NaN  For Sale   
1  https://iqibd.com/properties/295         Y        Y          NaN  For Sale   
2  https://iqibd.com/properties/298         Y        Y          NaN  For Sale   
3  https://iqibd.com/properties/300         Y        Y          NaN  For Sale   
4  https://iqibd.com/properties/297         Y        Y          NaN  For Sale   

  swimming_pool tv_cable         unit wifi  year_built  
0           NaN        Y  Single Unit    Y      2019.0  
1           NaN        Y      9A & 9B  NaN      2023.0  
2           NaN        Y           4A    Y      2023.0  
3             Y        Y           C6    Y      2023.0  
4           NaN      NaN          NaN  NaN      2010.0

In [62]:
#Get the info on the dataset
iqibd_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 261 entries, 0 to 260
Data columns (total 33 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   air_conditioning  48 non-null     object 
 1   area              261 non-null    object 
 2   balcony           19 non-null     object 
 3   building_type     261 non-null    object 
 4   cctv              214 non-null    object 
 5   central_heating   5 non-null      object 
 6   elevator          253 non-null    object 
 7   emergency_stairs  183 non-null    object 
 8   fitness_center    20 non-null     object 
 9   floor             260 non-null    object 
 10  garage            199 non-null    float64
 11  garden            8 non-null      object 
 12  generator         257 non-null    object 
 13  government_gas    18 non-null     object 
 14  interior          261 non-null    object 
 15  land_size         192 non-null    object 
 16  location          261 non-null    object 
 1

The dataset has 261 rows and 33 columns.

### Convert the columns to the required datatypes

In [63]:
#Extracting the numerical value from the area column since the unit is sqft or ft2
iqibd_df['area']=iqibd_df.area.str.split(r'[a-zA-Z]').str[0]
iqibd_df['area'].head()

0    1380 
1    1470 
2    1500 
3    5100 
4     710 
Name: area, dtype: object

In [64]:
#Converting the datatype to float
iqibd_df['area']=iqibd_df['area'].astype('float')
iqibd_df['area'].dtype

dtype('float64')

In [65]:
#Get the unique values in building_type column
iqibd_df['building_type'].unique()

array(['Apartment', 'Commercial property'], dtype=object)

In [66]:
#Set a new column building_nature value as 'Residential' or 'Commercial' based on building_type
iqibd_df["building_nature"] = ""
for index,row in iqibd_df.iterrows():
    if row['building_type']=='Apartment':
        building_nature = 'Residential'
    elif row['building_type']=='Commercial property':
        building_nature = 'Commercial'    
    iqibd_df.at[index,'building_nature'] = building_nature

In [67]:
iqibd_df.head()

air_conditioning    area balcony building_type cctv central_heating  \
0              NaN  1380.0     NaN     Apartment    Y             NaN   
1              NaN  1470.0     NaN     Apartment  NaN             NaN   
2              NaN  1500.0     NaN     Apartment    Y             NaN   
3                Y  5100.0       Y     Apartment    Y               Y   
4              NaN   710.0     NaN     Apartment  NaN             NaN   

  elevator emergency_stairs fitness_center floor  garage garden generator  \
0        Y                Y            NaN   5th     1.0    NaN         Y   
1        Y                Y            NaN   9th     1.0    NaN         Y   
2        Y                Y            NaN   4th     1.0    NaN         Y   
3        Y                Y              Y   6th     4.0      Y         Y   
4      NaN                Y            NaN   5th     NaN    NaN         Y   

  government_gas      interior   land_size      location  num_balcony  \
0            NaN  Un-Furnished   3.5 Katha   Aftabnagar,          4.0   
1            NaN  Un-Furnished     5 Katha       Mirpur,          3.0   
2              Y  Un-Furnished     3 Katha  Mohammadpur,          4.0   
3              Y  Un-Furnished  11.5 Katha    Baridhara,          4.0   
4              Y  Un-Furnished    12 Katha  Mohammadpur,          2.0   

   num_bath_rooms  num_bed_rooms parking pets_allow           price  \
0             3.0            3.0       Y        NaN     ৳9,800,000    
1             3.0            3.0       Y        NaN     ৳9,500,000    
2             4.0            3.0       Y        NaN    ৳10,500,000    
3             6.0            5.0       Y          Y   ৳139,150,000    
4             2.0            2.0     NaN        NaN     ৳4,500,000    

                      property_link reception security servant_room    status  \
0  https://iqibd.com/properties/306         Y        Y          NaN  For Sale   
1  https://iqibd.com/properties/295         Y        Y          NaN  For Sale   
2  https://iqibd.com/properties/298         Y        Y          NaN  For Sale   
3  https://iqibd.com/properties/300         Y        Y          NaN  For Sale   
4  https://iqibd.com/properties/297         Y        Y          NaN  For Sale   

  swimming_pool tv_cable         unit wifi  year_built building_nature  
0           NaN        Y  Single Unit    Y      2019.0     Residential  
1           NaN        Y      9A & 9B  NaN      2023.0     Residential  
2           NaN        Y           4A    Y      2023.0     Residential  
3             Y        Y           C6    Y      2023.0     Residential  
4           NaN      NaN          NaN  NaN      2010.0     Residential

In [68]:
#Setting the country column as 'Bangladesh'
#iqibd_df["country"] = "Bangladesh"
iqibd_df["city"] = iqibd_df.location.str.split(',').str[1]
iqibd_df["address"] = iqibd_df.location.str.split(',').str[0]
#Drop the location column
iqibd_df=iqibd_df.drop(['location'],axis=1)
iqibd_df.head()

air_conditioning    area balcony building_type cctv central_heating  \
0              NaN  1380.0     NaN     Apartment    Y             NaN   
1              NaN  1470.0     NaN     Apartment  NaN             NaN   
2              NaN  1500.0     NaN     Apartment    Y             NaN   
3                Y  5100.0       Y     Apartment    Y               Y   
4              NaN   710.0     NaN     Apartment  NaN             NaN   

  elevator emergency_stairs fitness_center floor  garage garden generator  \
0        Y                Y            NaN   5th     1.0    NaN         Y   
1        Y                Y            NaN   9th     1.0    NaN         Y   
2        Y                Y            NaN   4th     1.0    NaN         Y   
3        Y                Y              Y   6th     4.0      Y         Y   
4      NaN                Y            NaN   5th     NaN    NaN         Y   

  government_gas      interior   land_size  num_balcony  num_bath_rooms  \
0            NaN  Un-Furnished   3.5 Katha          4.0             3.0   
1            NaN  Un-Furnished     5 Katha          3.0             3.0   
2              Y  Un-Furnished     3 Katha          4.0             4.0   
3              Y  Un-Furnished  11.5 Katha          4.0             6.0   
4              Y  Un-Furnished    12 Katha          2.0             2.0   

   num_bed_rooms parking pets_allow           price  \
0            3.0       Y        NaN     ৳9,800,000    
1            3.0       Y        NaN     ৳9,500,000    
2            3.0       Y        NaN    ৳10,500,000    
3            5.0       Y          Y   ৳139,150,000    
4            2.0     NaN        NaN     ৳4,500,000    

                      property_link reception security servant_room    status  \
0  https://iqibd.com/properties/306         Y        Y          NaN  For Sale   
1  https://iqibd.com/properties/295         Y        Y          NaN  For Sale   
2  https://iqibd.com/properties/298         Y        Y          NaN  For Sale   
3  https://iqibd.com/properties/300         Y        Y          NaN  For Sale   
4  https://iqibd.com/properties/297         Y        Y          NaN  For Sale   

  swimming_pool tv_cable         unit wifi  year_built building_nature city  \
0           NaN        Y  Single Unit    Y      2019.0     Residential        
1           NaN        Y      9A & 9B  NaN      2023.0     Residential        
2           NaN        Y           4A    Y      2023.0     Residential        
3             Y        Y           C6    Y      2023.0     Residential        
4           NaN      NaN          NaN  NaN      2010.0     Residential        

       address  
0   Aftabnagar  
1       Mirpur  
2  Mohammadpur  
3    Baridhara  
4  Mohammadpur

In [69]:
#Fill the null values with 0 for num_bed_rooms and num_bath_rooms
iqibd_df['num_bed_rooms']=iqibd_df.num_bed_rooms.fillna(0)
iqibd_df['num_bath_rooms']=iqibd_df.num_bath_rooms.fillna(0)

In [70]:
#Convert num_bath_rooms and num_bed_rooms dtype as int from float
iqibd_df=iqibd_df.astype({'num_bath_rooms':'int','num_bed_rooms':'int'})
iqibd_df.head()

air_conditioning    area balcony building_type cctv central_heating  \
0              NaN  1380.0     NaN     Apartment    Y             NaN   
1              NaN  1470.0     NaN     Apartment  NaN             NaN   
2              NaN  1500.0     NaN     Apartment    Y             NaN   
3                Y  5100.0       Y     Apartment    Y               Y   
4              NaN   710.0     NaN     Apartment  NaN             NaN   

  elevator emergency_stairs fitness_center floor  garage garden generator  \
0        Y                Y            NaN   5th     1.0    NaN         Y   
1        Y                Y            NaN   9th     1.0    NaN         Y   
2        Y                Y            NaN   4th     1.0    NaN         Y   
3        Y                Y              Y   6th     4.0      Y         Y   
4      NaN                Y            NaN   5th     NaN    NaN         Y   

  government_gas      interior   land_size  num_balcony  num_bath_rooms  \
0            NaN  Un-Furnished   3.5 Katha          4.0               3   
1            NaN  Un-Furnished     5 Katha          3.0               3   
2              Y  Un-Furnished     3 Katha          4.0               4   
3              Y  Un-Furnished  11.5 Katha          4.0               6   
4              Y  Un-Furnished    12 Katha          2.0               2   

   num_bed_rooms parking pets_allow           price  \
0              3       Y        NaN     ৳9,800,000    
1              3       Y        NaN     ৳9,500,000    
2              3       Y        NaN    ৳10,500,000    
3              5       Y          Y   ৳139,150,000    
4              2     NaN        NaN     ৳4,500,000    

                      property_link reception security servant_room    status  \
0  https://iqibd.com/properties/306         Y        Y          NaN  For Sale   
1  https://iqibd.com/properties/295         Y        Y          NaN  For Sale   
2  https://iqibd.com/properties/298         Y        Y          NaN  For Sale   
3  https://iqibd.com/properties/300         Y        Y          NaN  For Sale   
4  https://iqibd.com/properties/297         Y        Y          NaN  For Sale   

  swimming_pool tv_cable         unit wifi  year_built building_nature city  \
0           NaN        Y  Single Unit    Y      2019.0     Residential        
1           NaN        Y      9A & 9B  NaN      2023.0     Residential        
2           NaN        Y           4A    Y      2023.0     Residential        
3             Y        Y           C6    Y      2023.0     Residential        
4           NaN      NaN          NaN  NaN      2010.0     Residential        

       address  
0   Aftabnagar  
1       Mirpur  
2  Mohammadpur  
3    Baridhara  
4  Mohammadpur

In [71]:
#Extract the price value alone in price column 
iqibd_df['price'] = iqibd_df['price'].str.extract(r'(\d.*)')
#Replace the commas in price string and convert dtype as float
iqibd_df['price']=iqibd_df['price'].str.replace(',','').astype('float')
iqibd_df['price'].head()

0      9800000.0
1      9500000.0
2     10500000.0
3    139150000.0
4      4500000.0
Name: price, dtype: float64

In [72]:
#Adding new columns property_description,property_overview and setting as null since values not available
iqibd_df["property_description"] = np.nan
iqibd_df["property_overview"] = np.nan

In [73]:
#Renaming the Status column as purpose
iqibd_df.rename(columns={'status':'purpose'},inplace=True)
iqibd_df.purpose.value_counts()

For Sale    261
Name: purpose, dtype: int64

In [74]:
#Set the value 'For Sale' as 'Sale' as only sale data is available in the site
iqibd_df['purpose']='Sale'
iqibd_df.head()

air_conditioning    area balcony building_type cctv central_heating  \
0              NaN  1380.0     NaN     Apartment    Y             NaN   
1              NaN  1470.0     NaN     Apartment  NaN             NaN   
2              NaN  1500.0     NaN     Apartment    Y             NaN   
3                Y  5100.0       Y     Apartment    Y               Y   
4              NaN   710.0     NaN     Apartment  NaN             NaN   

  elevator emergency_stairs fitness_center floor  garage garden generator  \
0        Y                Y            NaN   5th     1.0    NaN         Y   
1        Y                Y            NaN   9th     1.0    NaN         Y   
2        Y                Y            NaN   4th     1.0    NaN         Y   
3        Y                Y              Y   6th     4.0      Y         Y   
4      NaN                Y            NaN   5th     NaN    NaN         Y   

  government_gas      interior   land_size  num_balcony  num_bath_rooms  \
0            NaN  Un-Furnished   3.5 Katha          4.0               3   
1            NaN  Un-Furnished     5 Katha          3.0               3   
2              Y  Un-Furnished     3 Katha          4.0               4   
3              Y  Un-Furnished  11.5 Katha          4.0               6   
4              Y  Un-Furnished    12 Katha          2.0               2   

   num_bed_rooms parking pets_allow        price  \
0              3       Y        NaN    9800000.0   
1              3       Y        NaN    9500000.0   
2              3       Y        NaN   10500000.0   
3              5       Y          Y  139150000.0   
4              2     NaN        NaN    4500000.0   

                      property_link reception security servant_room purpose  \
0  https://iqibd.com/properties/306         Y        Y          NaN    Sale   
1  https://iqibd.com/properties/295         Y        Y          NaN    Sale   
2  https://iqibd.com/properties/298         Y        Y          NaN    Sale   
3  https://iqibd.com/properties/300         Y        Y          NaN    Sale   
4  https://iqibd.com/properties/297         Y        Y          NaN    Sale   

  swimming_pool tv_cable         unit wifi  year_built building_nature city  \
0           NaN        Y  Single Unit    Y      2019.0     Residential        
1           NaN        Y      9A & 9B  NaN      2023.0     Residential        
2           NaN        Y           4A    Y      2023.0     Residential        
3             Y        Y           C6    Y      2023.0     Residential        
4           NaN      NaN          NaN  NaN      2010.0     Residential        

       address  property_description  property_overview  
0   Aftabnagar                   NaN                NaN  
1       Mirpur                   NaN                NaN  
2  Mohammadpur                   NaN                NaN  
3    Baridhara                   NaN                NaN  
4  Mohammadpur                   NaN                NaN

In [75]:
iqibd_df.columns

Index(['air_conditioning', 'area', 'balcony', 'building_type', 'cctv',
       'central_heating', 'elevator', 'emergency_stairs', 'fitness_center',
       'floor', 'garage', 'garden', 'generator', 'government_gas', 'interior',
       'land_size', 'num_balcony', 'num_bath_rooms', 'num_bed_rooms',
       'parking', 'pets_allow', 'price', 'property_link', 'reception',
       'security', 'servant_room', 'purpose', 'swimming_pool', 'tv_cable',
       'unit', 'wifi', 'year_built', 'building_nature', 'city', 'address',
       'property_description', 'property_overview'],
      dtype='object')

In [76]:
#Fill the amenities column with 'N' for empty values
amenities_list=['air_conditioning','balcony','cctv','central_heating', 'elevator', 'emergency_stairs', 'fitness_center',
                'garden', 'generator', 'government_gas', 'parking','pets_allow', 'reception','security', 'servant_room',
                'swimming_pool', 'tv_cable', 'wifi']
#iqibd_df[amenities_list]=iqibd_df[amenities_list].fillna('N')
iqibd_df.head()

air_conditioning    area balcony building_type cctv central_heating  \
0              NaN  1380.0     NaN     Apartment    Y             NaN   
1              NaN  1470.0     NaN     Apartment  NaN             NaN   
2              NaN  1500.0     NaN     Apartment    Y             NaN   
3                Y  5100.0       Y     Apartment    Y               Y   
4              NaN   710.0     NaN     Apartment  NaN             NaN   

  elevator emergency_stairs fitness_center floor  garage garden generator  \
0        Y                Y            NaN   5th     1.0    NaN         Y   
1        Y                Y            NaN   9th     1.0    NaN         Y   
2        Y                Y            NaN   4th     1.0    NaN         Y   
3        Y                Y              Y   6th     4.0      Y         Y   
4      NaN                Y            NaN   5th     NaN    NaN         Y   

  government_gas      interior   land_size  num_balcony  num_bath_rooms  \
0            NaN  Un-Furnished   3.5 Katha          4.0               3   
1            NaN  Un-Furnished     5 Katha          3.0               3   
2              Y  Un-Furnished     3 Katha          4.0               4   
3              Y  Un-Furnished  11.5 Katha          4.0               6   
4              Y  Un-Furnished    12 Katha          2.0               2   

   num_bed_rooms parking pets_allow        price  \
0              3       Y        NaN    9800000.0   
1              3       Y        NaN    9500000.0   
2              3       Y        NaN   10500000.0   
3              5       Y          Y  139150000.0   
4              2     NaN        NaN    4500000.0   

                      property_link reception security servant_room purpose  \
0  https://iqibd.com/properties/306         Y        Y          NaN    Sale   
1  https://iqibd.com/properties/295         Y        Y          NaN    Sale   
2  https://iqibd.com/properties/298         Y        Y          NaN    Sale   
3  https://iqibd.com/properties/300         Y        Y          NaN    Sale   
4  https://iqibd.com/properties/297         Y        Y          NaN    Sale   

  swimming_pool tv_cable         unit wifi  year_built building_nature city  \
0           NaN        Y  Single Unit    Y      2019.0     Residential        
1           NaN        Y      9A & 9B  NaN      2023.0     Residential        
2           NaN        Y           4A    Y      2023.0     Residential        
3             Y        Y           C6    Y      2023.0     Residential        
4           NaN      NaN          NaN  NaN      2010.0     Residential        

       address  property_description  property_overview  
0   Aftabnagar                   NaN                NaN  
1       Mirpur                   NaN                NaN  
2  Mohammadpur                   NaN                NaN  
3    Baridhara                   NaN                NaN  
4  Mohammadpur                   NaN                NaN

In [77]:
#Rename the amenities columns for uniformity
amenity_dict={}
for amenity in amenities_list:
    amenity_dict.update({amenity:amenity+'-amenity'})
amenity_dict    

{'air_conditioning': 'air_conditioning-amenity',
 'balcony': 'balcony-amenity',
 'cctv': 'cctv-amenity',
 'central_heating': 'central_heating-amenity',
 'elevator': 'elevator-amenity',
 'emergency_stairs': 'emergency_stairs-amenity',
 'fitness_center': 'fitness_center-amenity',
 'garden': 'garden-amenity',
 'generator': 'generator-amenity',
 'government_gas': 'government_gas-amenity',
 'parking': 'parking-amenity',
 'pets_allow': 'pets_allow-amenity',
 'reception': 'reception-amenity',
 'security': 'security-amenity',
 'servant_room': 'servant_room-amenity',
 'swimming_pool': 'swimming_pool-amenity',
 'tv_cable': 'tv_cable-amenity',
 'wifi': 'wifi-amenity'}

In [78]:
iqibd_df.rename(columns=amenity_dict,inplace=True)
iqibd_df.head()

air_conditioning-amenity    area balcony-amenity building_type cctv-amenity  \
0                      NaN  1380.0             NaN     Apartment            Y   
1                      NaN  1470.0             NaN     Apartment          NaN   
2                      NaN  1500.0             NaN     Apartment            Y   
3                        Y  5100.0               Y     Apartment            Y   
4                      NaN   710.0             NaN     Apartment          NaN   

  central_heating-amenity elevator-amenity emergency_stairs-amenity  \
0                     NaN                Y                        Y   
1                     NaN                Y                        Y   
2                     NaN                Y                        Y   
3                       Y                Y                        Y   
4                     NaN              NaN                        Y   

  fitness_center-amenity floor  garage garden-amenity generator-amenity  \
0                    NaN   5th     1.0            NaN                 Y   
1                    NaN   9th     1.0            NaN                 Y   
2                    NaN   4th     1.0            NaN                 Y   
3                      Y   6th     4.0              Y                 Y   
4                    NaN   5th     NaN            NaN                 Y   

  government_gas-amenity      interior   land_size  num_balcony  \
0                    NaN  Un-Furnished   3.5 Katha          4.0   
1                    NaN  Un-Furnished     5 Katha          3.0   
2                      Y  Un-Furnished     3 Katha          4.0   
3                      Y  Un-Furnished  11.5 Katha          4.0   
4                      Y  Un-Furnished    12 Katha          2.0   

   num_bath_rooms  num_bed_rooms parking-amenity pets_allow-amenity  \
0               3              3               Y                NaN   
1               3              3               Y                NaN   
2               4              3               Y                NaN   
3               6              5               Y                  Y   
4               2              2             NaN                NaN   

         price                     property_link reception-amenity  \
0    9800000.0  https://iqibd.com/properties/306                 Y   
1    9500000.0  https://iqibd.com/properties/295                 Y   
2   10500000.0  https://iqibd.com/properties/298                 Y   
3  139150000.0  https://iqibd.com/properties/300                 Y   
4    4500000.0  https://iqibd.com/properties/297                 Y   

  security-amenity servant_room-amenity purpose swimming_pool-amenity  \
0                Y                  NaN    Sale                   NaN   
1                Y                  NaN    Sale                   NaN   
2                Y                  NaN    Sale                   NaN   
3                Y                  NaN    Sale                     Y   
4                Y                  NaN    Sale                   NaN   

  tv_cable-amenity         unit wifi-amenity  year_built building_nature city  \
0                Y  Single Unit            Y      2019.0     Residential        
1                Y      9A & 9B          NaN      2023.0     Residential        
2                Y           4A            Y      2023.0     Residential        
3                Y           C6            Y      2023.0     Residential        
4              NaN          NaN          NaN      2010.0     Residential        

       address  property_description  property_overview  
0   Aftabnagar                   NaN                NaN  
1       Mirpur                   NaN                NaN  
2  Mohammadpur                   NaN                NaN  
3    Baridhara                   NaN                NaN  
4  Mohammadpur                   NaN                NaN

In [79]:
#Replace all the 'Y' values to yes for the amenities columns
amenity_list=list(amenity_dict.values())
iqibd_df[amenity_list]=iqibd_df[amenity_list].replace({'Y':'yes'})
iqibd_df.head()

air_conditioning-amenity    area balcony-amenity building_type cctv-amenity  \
0                      NaN  1380.0             NaN     Apartment          yes   
1                      NaN  1470.0             NaN     Apartment          NaN   
2                      NaN  1500.0             NaN     Apartment          yes   
3                      yes  5100.0             yes     Apartment          yes   
4                      NaN   710.0             NaN     Apartment          NaN   

  central_heating-amenity elevator-amenity emergency_stairs-amenity  \
0                     NaN              yes                      yes   
1                     NaN              yes                      yes   
2                     NaN              yes                      yes   
3                     yes              yes                      yes   
4                     NaN              NaN                      yes   

  fitness_center-amenity floor  garage garden-amenity generator-amenity  \
0                    NaN   5th     1.0            NaN               yes   
1                    NaN   9th     1.0            NaN               yes   
2                    NaN   4th     1.0            NaN               yes   
3                    yes   6th     4.0            yes               yes   
4                    NaN   5th     NaN            NaN               yes   

  government_gas-amenity      interior   land_size  num_balcony  \
0                    NaN  Un-Furnished   3.5 Katha          4.0   
1                    NaN  Un-Furnished     5 Katha          3.0   
2                    yes  Un-Furnished     3 Katha          4.0   
3                    yes  Un-Furnished  11.5 Katha          4.0   
4                    yes  Un-Furnished    12 Katha          2.0   

   num_bath_rooms  num_bed_rooms parking-amenity pets_allow-amenity  \
0               3              3             yes                NaN   
1               3              3             yes                NaN   
2               4              3             yes                NaN   
3               6              5             yes                yes   
4               2              2             NaN                NaN   

         price                     property_link reception-amenity  \
0    9800000.0  https://iqibd.com/properties/306               yes   
1    9500000.0  https://iqibd.com/properties/295               yes   
2   10500000.0  https://iqibd.com/properties/298               yes   
3  139150000.0  https://iqibd.com/properties/300               yes   
4    4500000.0  https://iqibd.com/properties/297               yes   

  security-amenity servant_room-amenity purpose swimming_pool-amenity  \
0              yes                  NaN    Sale                   NaN   
1              yes                  NaN    Sale                   NaN   
2              yes                  NaN    Sale                   NaN   
3              yes                  NaN    Sale                   yes   
4              yes                  NaN    Sale                   NaN   

  tv_cable-amenity         unit wifi-amenity  year_built building_nature city  \
0              yes  Single Unit          yes      2019.0     Residential        
1              yes      9A & 9B          NaN      2023.0     Residential        
2              yes           4A          yes      2023.0     Residential        
3              yes           C6          yes      2023.0     Residential        
4              NaN          NaN          NaN      2010.0     Residential        

       address  property_description  property_overview  
0   Aftabnagar                   NaN                NaN  
1       Mirpur                   NaN                NaN  
2  Mohammadpur                   NaN                NaN  
3    Baridhara                   NaN                NaN  
4  Mohammadpur                   NaN                NaN

In [80]:
#Renaming the columns again for uniformity
iqibd_df.rename(columns={'balcony-amenity':'balcony-or-terrace-amenity','cctv-amenity':'cctv-security-amenity',
                        'central_heating-amenity':'central-heating-amenity','elevator-amenity':'elevators-in-building-amenity',
                        'garden-amenity':'lawn-or-garden-amenity','parking-amenity':'parking-spaces-amenity',
                        'property_link':'property_url','swimming_pool-amenity':'swimming-pool-amenity',
                        },inplace=True)

In [81]:
iqibd_df.columns

Index(['air_conditioning-amenity', 'area', 'balcony-or-terrace-amenity',
       'building_type', 'cctv-security-amenity', 'central-heating-amenity',
       'elevators-in-building-amenity', 'emergency_stairs-amenity',
       'fitness_center-amenity', 'floor', 'garage', 'lawn-or-garden-amenity',
       'generator-amenity', 'government_gas-amenity', 'interior', 'land_size',
       'num_balcony', 'num_bath_rooms', 'num_bed_rooms',
       'parking-spaces-amenity', 'pets_allow-amenity', 'price', 'property_url',
       'reception-amenity', 'security-amenity', 'servant_room-amenity',
       'purpose', 'swimming-pool-amenity', 'tv_cable-amenity', 'unit',
       'wifi-amenity', 'year_built', 'building_nature', 'city', 'address',
       'property_description', 'property_overview'],
      dtype='object')

In [83]:
# Check for duplicates
# dup=iqibd_df[iqibd_df.duplicated(keep='first')]
# dup

In [85]:
# Save cleaned dataset to csv
iqibd_df.to_csv("cleaned/cleaned_iqibd_properties.csv", index=False)